sentry/src/main/java/io/sentry/SentryTransaction.java

```diff
@@ -0,0 +1,218 @@
+    Objects.requireNonNull(contexts, "contexts are required");
+    this.transaction = Objects.requireNonNull(name, "name is required");
+    this.startTimestamp = DateUtils.getCurrentDateTime();
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+    Contexts ctx = new Contexts();
+    ctx.setTrace(contexts);
+    this.setContexts(ctx);
+  }
+
+  /**
+   * Sets transaction name.
+   *
+   * @param name - transaction name
+   */
+  void setName(final @NotNull String name) {
+    Objects.requireNonNull(name, "name is required");
+    this.transaction = name;
+  }
+
+  /**
+   * Starts a child Span.
+   *
+   * @return a new transaction span
+   */
+  @Override
+  public Span startChild() {
+    return this.startChild(getSpanId());
+  }
+
+  /**
+   * Starts a child Span with given trace id and parent span id.
+   *
+   * @param parentSpanId - parent span id
+   * @return a new transaction span
+   */
+  Span startChild(final @NotNull SpanId parentSpanId) {
+    Objects.requireNonNull(parentSpanId, "parentSpanId is required");
+    final Span span = new Span(getTraceId(), parentSpanId, this);
+    this.spans.add(span);
+    return span;
+  }
+
+  @Override
+  public SentryTraceHeader toSentryTrace() {
+    return new SentryTraceHeader(getTraceId(), getSpanId(), isSampled());
+  }
+
+  @NotNull
+  SpanId getSpanId() {
+    return getContexts().getTrace().getSpanId();
+  }
+
+  @NotNull
+  SentryId getTraceId() {
+    return getContexts().getTrace().getTraceId();
+  }
+
+  @Nullable
+  Boolean isSampled() {
+    return getContexts().getTrace().getSampled();
+  }
+
+  @Override
+  public void finish() {
+    this.timestamp = DateUtils.getCurrentDateTime();
+    this.hub.captureTransaction(this, null);
+  }
+
+  /**
+   * Sets transaction operation.
+   *
+   * @param op - operation
+   */
+  @Override
+  public void setOp(@Nullable String op) {
+    this.getContexts().getTrace().setOp(op);
+  }
+
+  /**
+   * Sets transaction description.
+   *
+   * @param description - the description.
+   */
+  @Override
+  public void setDescription(@Nullable String description) {
+    this.getContexts().getTrace().setDescription(description);
+  }
+
+  public @Nullable String getDescription() {
+    return this.getContexts().getTrace().getDescription();
+  }
+
+  @Override
+  public @NotNull SpanContext getSpanContext() {
+    return this.getContexts().getTrace();
+  }
+
+  /**
+   * Sets transaction status.
+   *
+   * @param spanStatus - the status
+   */
+  @Override
+  public void setStatus(@Nullable SpanStatus spanStatus) {
+    this.getContexts().getTrace().setStatus(spanStatus);
+  }
+
+  /**
+   * Returns the transaction name.
+   *
+   * @return transaction name.
+   */
+  @Nullable
+  String getTransaction() {
+    return transaction;
+  }
+
+  @NotNull
+  Date getStartTimestamp() {
+    return startTimestamp;
+  }
+
+  @Nullable
+  Date getTimestamp() {
+    return timestamp;
+  }
+
+  @NotNull
+  Collection<Span> getSpans() {
```

Comment:
can it be a `List`? or is there's reason to be a `Collection`?

Reply:
👍